### ISSUE TITLE
[nodes-from-anchor] Handle generic types

### ISSUE THREAD

**Issue by ngundotra** [codama-idl/codama#308]
While rendering JS client for IDL for `solana program dump vmT2hAx4N2U6DyjYxgQHER4VGC8tHJCfHNsSepBKCJZ`,  

I got the following: `CodamaError: Unrecognized Anchor IDL type [{"generic":"T"}].`

```javascript
import { renderVisitor } from '@codama/renderers-js';
import { createFromRoot } from 'codama';
import { rootNodeFromAnchor } from '@codama/nodes-from-anchor';

import fs from 'fs';
import path from 'path';

// solana program dump vmT2hAx4N2U6DyjYxgQHER4VGC8tHJCfHNsSepBKCJZ
const anchorIdl = JSON.parse(fs.readFileSync('anchor-idl.json', 'utf8'));
const codama = createFromRoot(rootNodeFromAnchor(anchorIdl));

const pathToGeneratedFolder = path.join(__dirname, 'clients', 'rust', 'src', 'generated');
const options = {}; // See below.
codama.accept(renderVisitor(pathToGeneratedFolder, options));

```

**Comment by ngundotra** [comment]
Can get the Anchor IDL from here: https://explorer.solana.com/address/vmT2hAx4N2U6DyjYxgQHER4VGC8tHJCfHNsSepBKCJZ/anchor-program

**Comment by lorisleiva** [comment]
Oh yeah that's because Codama doesn't support generic types. I thought there was an issue for this already but I couldn't find it.

The main reason Codama doesn't support generics (yet?) is because the Codama spec aims to be language agnostic. It describes high-level program logic and binary layouts regardless of how the program was written. It also means it can be used to render client code for any languages. Many of whom don't support generics. So by adding generics support to the spec, we are limiting the set of languages Codama can work with.

Anchor doesn't need to worry about this because it limits itself to Rust on the program side and JavaScript on the client side.

A potential solution to this would be to force a "fallback" type for generic types that can be used when generics are not supported by the language. But this wouldn't help here with the conversion from Anchor.

**Comment by ngundotra** [comment]
Can we get a fallback type for now so anchor users can at least use non-generic aspects of their IDL?

**Comment by lorisleiva** [comment]
Do we have any relevant info on the Anchor IDL to provide a fallback type? Like if I see a generic type `T`, what should I transform it to?

**Comment by AsheR4444** [comment]
@ngundotra did you found any solutions ?

**Comment by ngundotra** [comment]
No, haven't worked on this in awhile. Happy to work on one if you need it @AsheR4444 !
